backend/app/main.py

```diff
@@ -0,0 +1,158 @@
+from typing import Optional
+
+import httpx
+import jwt
+from authlib.common.urls import add_params_to_uri
+from authlib.integrations.starlette_client import OAuth, OAuthError
+from fastapi import Cookie, FastAPI, HTTPException, Query, Request
+from fastapi.responses import JSONResponse, RedirectResponse
+from fastapi.routing import APIRouter
+from httpx import Timeout
+from pyjwt_key_fetcher import AsyncKeyFetcher
+from starlette.middleware.sessions import SessionMiddleware
+
+from .settings import conf
+
+app = FastAPI()
+app.add_middleware(SessionMiddleware, secret_key=conf.SESSION_SECRET)
+router = APIRouter()
+
+oauth = OAuth()
+oauth.register(
+    name="login_portal",
+    client_id=conf.OIDC_CLIENT_ID,
+    client_secret=conf.OIDC_CLIENT_SECRET,
+    server_metadata_url=(conf.OIDC_PROVIDER_URL + "/.well-known/openid-configuration"),
+    client_kwargs={
+        "scope": conf.OIDC_SCOPES,
+        "timeout": Timeout(timeout=conf.OIDC_REQUEST_TIMEOUT),
+    },
+)
+
+key_fetcher = AsyncKeyFetcher()
+
+
+@router.get("/login")
+async def login(request: Request):
```

Comment:
We could maybe add some minor docstring to`/login`, `/auth` and `/logout` to explain they're just some basic stuff for handling the OpenID Connect login/logout flows and using the `Authlib` library to do it.